carbonserver/carbonserver/api/infra/repositories/repository_users.py

```diff
@@ -0,0 +1,56 @@
+import hashlib
+import os
+import secrets
+from typing import List
+
+from pydantic import SecretStr
+
+from carbonserver.api import schemas
+from carbonserver.api.domain.user import User
+from carbonserver.database import models
+
+
+class InMemoryRepository(User):
+    def __init__(self):
+        self.users: List[User] = []
+        self.id: int = 0
+        self.inactive_users: List = []
+
+    def create_user(self, user: schemas.UserCreate) -> str:
+        self.id += 1
+        self.users.append(
+            models.User(
+                user_id=self.id,
+                name=user.name,
+                email=user.email,
+                hashed_password=_hash_password(user.password),
+                api_key=_api_key_generator(),
+                is_active=user.is_active,
+            )
+        )
+        return self.users[self.id - 1]
+
+    def get_user_by_id(self, user_id: int):
+        user = [user for user in self.users if user.user_id == user_id][0]
+        return user
+
+    def list_users(self):
+        user_ids = [user.user_id for user in self.users]
+        return user_ids
+
+
+def _hash_password(password: str) -> SecretStr:
+    salt = _random_generator()  # Extract random generation to make it easier to mock
```

Comment:
How can you check the password of the user with a random salt when he will log in ?
We have to store the salt in the database, with the user's information, isn't it ?

Reply:
Indeed, the salt should be stored too. However this implementation would require a lot of User internal methods to check the password validity (request salt + hashed key from db, hash the user input and compare results) or to update it.
I found another implementation that seems interesting : https://variable-scope.com/posts/storing-and-verifying-passwords-with-sqlalchemy. We could first use secret passwords for the first version, and try to implement this authentication logic after. Wdyt ?